src/navigation/OnboardingStack.tsx

```diff
@@ -23,32 +23,59 @@ const onboardingScreenOptions: StackNavigationOptions = {
   headerShown: false,
 }
 
-const OnboardingStack: FunctionComponent = () => {
+interface OnboardingStackProps {
+  destinationOnSkip: DestinationStack
+}
+
+const OnboardingStack: FunctionComponent<OnboardingStackProps> = ({
+  destinationOnSkip,
+}) => {
+  const onboardingScreens = useOnboardingData(destinationOnSkip)
   return (
     <Stack.Navigator screenOptions={onboardingScreenOptions}>
-      <Stack.Screen name={OnboardingScreens.Welcome} component={Welcome} />
-      <Stack.Screen
-        name={OnboardingScreens.Introduction}
-        component={Introduction}
-      />
-      <Stack.Screen
-        name={OnboardingScreens.PhoneRemembersDevices}
-        component={PhoneRemembersDevices}
-      />
-      <Stack.Screen
-        name={OnboardingScreens.PersonalPrivacy}
-        component={PersonalPrivacy}
-      />
-      <Stack.Screen
-        name={OnboardingScreens.GetNotified}
-        component={GetNotified}
-      />
-      <Stack.Screen
-        name={OnboardingScreens.ValueProposition}
-        component={ValueProposition}
-      />
+      <Stack.Screen name={Screens.Welcome} component={Welcome} />
+      {onboardingScreens.map(
```

Comment:
```suggestion
      const toStackScreen = (data) => Component
      {onboardingScreenData.map(toStackScreen)
```

I like this refactor with the hook and fewer component files a lot! better separation of data from markup and its nice to have the data all in one spot

Reply:
Heck yaaaaaa

Reply:
I like this `toStackScreen` function, very Elm-y.